.sphinx/_static/furo_colors.css

```diff
@@ -28,6 +28,7 @@ body {
     --color-highlighted-background: #EBEBEB;
     --color-link-underline: var(--color-link);
     --color-link-underline--hover: var(--color-link);
+    --color-link-underline--visited: var(--color-link--visited);
```

Comment:
You'll also need `--color-link-underline--visited--hover: var(--color-link--visited);` - otherwise we have different colours for muted links (see the links in the footer).